corehq/apps/hqcase/api/get_bulk.py

```diff
@@ -0,0 +1,86 @@
+import dataclasses
+from dataclasses import dataclass, field
+
+from corehq.apps.es.case_search import CaseSearchES, ElasticCaseSearch
+from corehq.apps.hqcase.api.core import UserError, serialize_es_case
+from corehq.apps.hqcase.api.get_list import MAX_PAGE_SIZE
+
+
+@dataclass
+class BulkFetchResults:
+    cases: list = field(default_factory=list)
+    matching_records: int = 0
+    missing_records: int = 0
+
+    def merge(self, results):
+        self.cases.extend(results.cases)
+        self.matching_records += results.matching_records
+        self.missing_records += results.missing_records
+
+
+def get_bulk(domain, case_ids=None, external_ids=None):
+    """Get cases in bulk.
+
+    This must return a result for each case ID passed in and the results must
+    be in the same order as the original list of case IDs.
+
+    If both case IDs and external IDs are passed then results will include
+    cases loaded by ID first followed by cases loaded by external ID.
+
+    If a case is not found or belongs to a different domain then
+    an error stub is included in the result set.
+    """
+    case_ids = case_ids or []
+    external_ids = external_ids or []
+    if len(case_ids) + len(external_ids) > MAX_PAGE_SIZE:
+        raise UserError(f"You cannot request more than {MAX_PAGE_SIZE} cases per request.")
+
+    results = BulkFetchResults()
+    if case_ids:
+        results.merge(_get_cases_by_id(domain, case_ids))
+
+    if external_ids:
+        results.merge(_get_cases_by_external_id(domain, external_ids))
+
+    return dataclasses.asdict(results)
+
+
+def _get_cases_by_id(domain, case_ids):
+    es_results = ElasticCaseSearch().get_docs(case_ids)
```

Comment:
aside: I'm thinking about switching individual gets to use this same method (they pull from SQL at the moment)